### DISCUSSION TITLE
How to setup multiple strings?

### DISCUSSION THREAD

**Discussion by Captain0Kangaroo** [nathanmarlor/foxess_modbus#103] [Q&A] [answered]
Would like to know the best way to setup PV1 and PV2 daily power to add into the Energy Config rather than using the Total Generation.

In the 'other' modbus version these were available to use but cant see them in this one.

**Comment by Captain0Kangaroo** [comment]
To add to this as i'm not sure if its the reason but i get no data for sensor.foxess_modbus_pv1_energy_total and sensor.foxess_modbus_pv2_energy_total

**Reply by canton7** [reply]
Did you give your inverter the friendly name foxess_modbus?

**Reply by Captain0Kangaroo** [reply]
> Did you give your inverter the friendly name foxess_modbus?

I don't think i used and underscore.. does that matter?

**Comment by canton7** [comment, marked as answer]
The latest betas provide pv1_energy_total / pv1_energy_total sensors

**Reply by Captain0Kangaroo** [reply]
> The latest betas provide pv1_energy_total / pv1_energy_total sensors

I get unknown from these sensors. I think this is the problem. (i'm on the latest beta)

<img width="562" alt="Screen Shot 2023-03-31 at 14 13 51" src="https://user-images.githubusercontent.com/96144685/229129855-19e7f3d8-65d0-44e2-82f9-2acba9a8d409.png">

**Reply by canton7** [reply]
Can you see in HA what the source sensor for the integration is (it's in one of the drop-downs iirc)? Is that correct, and does that sensor have a value?

**Reply by Captain0Kangaroo** [reply]
> Can you see in HA what the source sensor for the integration is (it's in one of the drop-downs iirc)? Is that correct, and does that sensor have a value?

Do you mean these?
<img width="307" alt="Screen Shot 2023-03-31 at 14 40 15" src="https://user-images.githubusercontent.com/96144685/229135866-6f2dad08-e329-442a-a478-757b843aef64.png">

**Reply by canton7** [reply]
![Screenshot_20230331-162037](https://user-images.githubusercontent.com/568104/229146734-8da5b77d-80ba-4bdf-8689-7ddf648cec88.jpg)

That bit, at the bottom, which says what the source sensor is.

**Reply by Captain0Kangaroo** [reply]
<img width="566" alt="Screen Shot 2023-03-31 at 15 54 25" src="https://user-images.githubusercontent.com/96144685/229155509-bb9e85fe-6156-4094-ab2d-6a06bb1a8e52.png">
This is what mine is....

**Reply by canton7** [reply]
Ah, that's the issue. @nathanmarlor?

**Reply by canton7** [reply]
Might be affecting the energy dashboard stuff as well?

**Reply by nathanmarlor** [reply]
Doh. Yep, will fix.

**Reply by Captain0Kangaroo** [reply]
> Doh. Yep, will fix.

Cool - - Was doing my head in as to why i couldn't get it working.

**Reply by Captain0Kangaroo** [reply]
I think the issue was because i gave it a name (didn't read the install notes). Deleted and added again (without a name) and it seems to be working now.

**Reply by Captain0Kangaroo** [reply]
Confirmed - - Giving it a name in the setup adds that name to the sensors which in may case was 'FoxESS Modbus' so all my sensors were prefixed with sensor.FoxESS Modbus_sensorname.

**Reply by canton7** [reply]
Tracked by #106
